src/idiomatic/leveraging-the-type-system/borrow-checker-invariants/phantomdata-04-borrowedfd.md

```diff
@@ -0,0 +1,94 @@
+---
+minutes: 10
+---
+
+# PhantomData 4/4: OwnedFd & BorrowedFd
+
+`BorrowedFd` is a prime example of `PhantomData` in action.
+
+```rust,editable
+use libc::{O_CREAT, O_WRONLY, close, open};
+use std::marker::PhantomData;
+use std::os::raw::c_int;
+
+struct OwnedFd {
+    fd: c_int,
+}
+
+impl OwnedFd {
+    fn try_from_fd(fd: c_int) -> Option<Self> {
+        if fd < 0 {
+            return None;
+        }
+        Some(OwnedFd { fd })
+    }
+
+    fn as_fd<'a>(&'a self) -> BorrowedFd<'a> {
+        BorrowedFd { fd: self.fd, _phantom: PhantomData }
+    }
+}
+
+impl Drop for OwnedFd {
+    fn drop(&mut self) {
+        unsafe { close(self.fd) };
+    }
+}
+
+struct BorrowedFd<'a> {
+    fd: c_int,
+    _phantom: PhantomData<&'a ()>,
+}
+
+fn main() {
+    // A file we're creating with a raw syscall. Imagine this comes from an FFI.
+    let fd = unsafe { open(c"c_str.txt".as_ptr(), O_WRONLY | O_CREAT) };
+    // Creating an ownedfd from a file, when this drops the file descriptor
+    // will be closed.
```

Comment:
```suggestion
    // Pass the ownership of an integer file descriptor to an `OwnedFd`.
    // `OwnedFd::drop()` closes the file descriptor.
```

Capitalization + rephrasing. I don't like the phrase "when this drops" (it sounds active like the object itself decides to drop, and that is not how it works).